feat: setting up of async redux

zrnkrystnlee opened the pull request:
- added app state and freezed it
- added a home page

Working homepage
![image](https://user-images.githubusercontent.com/72370369/225824537-3d4b37fc-f629-4259-9869-5163977f9aa9.png)

zrnkrystnlee committed:
fix: added a TODO Comment for constants
- added TODO Comment in title and Text

zrnkrystnlee committed:
feat: setting up of async redux
- added app state and freezed it
- added a home page

shawn-ffufm commented on pokedex_async_redux/lib/home_page.dart:
```diff
@@ -0,0 +1,24 @@
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text("Home Page"),
```
kindly add a todo comment that this will be placed in a constant later 🙂

shawn-ffufm commented on pokedex_async_redux/lib/home_page.dart:
```diff
@@ -0,0 +1,24 @@
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: const [
+            Text(
+              'This is the Home Page',
```
same here and no need for extra trailing comma 🙂

shawn-ffufm commented on pokedex_async_redux/lib/main.dart:
```diff
@@ -1,115 +1,22 @@
+    StoreProvider(
+      store: store,
+      child: const MaterialApp(
+        title: 'Home Page',
```
see related feedback above 🙂

shawn-ffufm reviewed (changes requested):
@zrnkrystnlee see minor pr feedback 🙂

zrnkrystnlee committed:
feature: added a TODO Comment for constants
- added TODO Comment in title and Text

zrnkrystnlee committed:
fix: added a TODO Comment for constants
- added TODO Comment in title and Text

zrnkrystnlee committed:
Merge branch 'feature/setup-async-redux' of https://github.com/zrnkrystnlee/pokedex-async-redux into feature/setup-async-redux

zrnkrystnlee committed:
feature: added a TODO Comment
- added TODO Comments on title of  main.dart and title and text in home_page.dart
